pkg/agent/controller/egress/egress_controller.go

```diff
@@ -223,8 +266,22 @@ func (c *EgressController) Run(stopCh <-chan struct{}) {
 		return
 	}
 
+	go c.cluster.Run(stopCh)
+
 	go wait.NonSlidingUntil(c.watchEgressGroup, 5*time.Second, stopCh)
 
+	// if node support egress failover
+	if c.cluster.LocalNodeJoined() {
+		// when a new node join gossip cluster, it should handle the egress that hit itself(moved from other nodes)
+		egresses, _ := c.egressLister.List(labels.Everything())
+
+		for _, e := range egresses {
+			if e.Spec.ExternalIPPool != "" && c.cluster.ShouldSelectEgress(e) {
+				c.enqueueEgress(e)
+			}
+		}
+	}
```

Comment:
Is this code block required? It runs once. What if localNodeJoined is not true? If it doesn't cause any problem. Then it seems we don't need to run it